### PR TITLE
test(private repo): migrate private repo tests from scylla-artifact-tests

### PR DESCRIPTION
**Pull request by enaydanov**
Trello: https://trello.com/c/mPsZOchp/1653-private-repo-artifact-test

## PR pre-checks (self review)

- [x] I followed [KISS principle](https://en.wikipedia.org/wiki/KISS_principle) and [best practices](https://docs.google.com/document/d/1jihgOKb5iGRlD8_HQ92O0JbLk1kASUoZT23i_MXFSKI)
- [x] I gave variables/functions meaningful self-explanatory names
- [x] I didn't leave commented-out/debugging code
- [x] I didn't copy-paste code
- [x] I added the relevant `backport` labels
- [ ] ~~New configuration option are added and documented (in `sdcm/sct_config.py`)~~
- [ ] ~~I have added tests to cover my changes (Infrastructure only - under `unit-test/` folder)~~
- [x] All new and existing unit tests passed (CI)
- [x] I have updated the Readme/doc folder accordingly (if needed)

### CODE REVIEW

File: test-cases/private-repo/private-repo.yaml [hunk 1]
```diff
@@ -0,0 +1,9 @@
+cluster_backend: 'gce'
+logs_transport: 'ssh'
+n_db_nodes: 0
+n_loaders: 0
+n_monitor_nodes: 0
+store_results_in_elasticsearch: False
+nemesis_class_name: 'NoOpMonkey'
+user_prefix: 'private-repo'
+scylla_repo: 'https://repositories.scylladb.com/scylla/repo/amos-test-centos/centos/scylladb-1.7.repo'
```
Comment (amoskong) [inline test-cases/private-repo/private-repo.yaml]: Let's rename the UUID `amos-test-centos` to `qa-test-centos`

Comment (enaydanov) [reply test-cases/private-repo/private-repo.yaml]: done

File: private_repo_test.py [hunk 2]
```diff
@@ -0,0 +1,183 @@
...
+def extract_pkginfo_urls_from_yum_repo(repofile):
+    urls = []
+    for line in repofile.splitlines():
+        if line.startswith("baseurl="):
+            # Convert string like
+            #   `baseurl=https://.../scylla/centos/scylladb-1.7/$releasever/$basearch/'
+            # to
+            #   `https://.../scylla/centos/scylladb-1.7/7Server/x86_64/repodata/repomd.xml'
+            urls.append(
+                line[8:].replace("$releasever", "7Server").replace("$basearch", "x86_64") + "repodata/repomd.xml")
```
Comment (amoskong) [inline private_repo_test.py]: From 3.3, the '7Server' directory is removed.   The packages support both CentOS7 & CentOS8.

- http://downloads.scylladb.com/downloads/scylla/rpm/centos/scylladb-3.2/7Server/x86_64/
- http://downloads.scylladb.com/downloads/scylla/rpm/centos/scylladb-3.3/x86_64/

Comment (enaydanov) [reply private_repo_test.py]: Good to know. But it's not a problem for this piece of code: if there is not `$releasever` in the line it'll not be replaced by `7Server`:

```sh
$ curl https://repositories.scylladb.com/scylla/repo/qa-test-centos/centos/scylladb-3.3.repo
[scylla-3.3]
name=Scylla for Centos - $basearch
baseurl=https://repositories.scylladb.com/scylla/downloads/scylladb/qa-test-centos/scylla/rpm/centos/scylladb-3.3/$basearch/
enabled=1
gpgcheck=0

[scylla-generic-3.3]
name=Scylla for centos
baseurl=https://repositories.scylladb.com/scylla/downloads/scylladb/qa-test-centos/scylla/rpm/centos/scylladb-3.3/noarch/
enabled=1
gpgcheck=0

```

File: sdcm/utils/housekeeping.py [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+DB_NAME = "housekeeping"
+
+LOGGER = logging.getLogger(__name__)
+
+
+Row = Sequence[Any]
+
+
+class HousekeepingDB:
+    def __init__(self, host: str, username: str, password: str, log: logging.Logger = LOGGER):
```
Comment (bentsi) [inline sdcm/utils/housekeeping.py]: let's use global logger

**Comment by amoskong** [comment]
Hi @enaydanov @bentsi ,

We need this PR for 3.3 for covering private-repo test of debian10 & centos8

**Pull request merged**